Fix for JavaScript module path to handle app not being hosted at the root of the website

victorperez2911 opened the pull request:
…rent from the WebSite Root (e.g. using the href attribute element of the index.html base element)

victorperez2911 committed:
Fix for calculating the Url Base in a wasm app served in a path different from the WebSite Root (e.g. using the href attribute element of the index.html base element)

tpeczek commented on src/Blazor.BroadcastChannel/BroadcastChannelService.cs:
```diff
@@ -1,14 +1,19 @@
-        public BroadcastChannelService(IJSRuntime jsRuntime)
+        public BroadcastChannelService(IJSRuntime jsRuntime, NavigationManager NavManager)
         {
-            _moduleTask = new(() => jsRuntime.InvokeAsync<IJSObjectReference>("import", "/_content/Blazor.BroadcastChannel/Blazor.BroadcastChannel.js"));
```
I've done some testing and it looks that it might be enough to change the path to `"./_content/Blazor.BroadcastChannel/Blazor.BroadcastChannel.js"`. That said, I don't know if I have reproduced your scenario, can you test that approach?

victorperez2911 replied:
The scenario I brought up only occurs after deployment, which makes testing your approach difficult.

Can you publish a pre-release of the package with this change?

tpeczek replied:
Sure thing, but it may need to wait till weekend.

tpeczek commented on src/Blazor.BroadcastChannel/BroadcastChannelService.cs:
```diff
@@ -1,14 +1,19 @@
+
+            string path = new Uri(NavManager.BaseUri).AbsolutePath + "_content/Blazor.BroadcastChannel/Blazor.BroadcastChannel.js";
+
+            //_moduleTask = new(() => jsRuntime.InvokeAsync<IJSObjectReference>("import", "/_content/Blazor.BroadcastChannel/Blazor.BroadcastChannel.js"));
```
No need to comment out the code, you can just delete it 😉.

victorperez2911 committed:
The most significant changes involve the modification of the `BroadcastChannelService` class constructor and the adjustment of how the `path` variable is set. The `NavigationManager` parameter has been removed from the constructor of the `BroadcastChannelService` class. The `path` variable, which was previously set using the `BaseUri` property of the `NavigationManager` instance, is now directly set to a specific string.

Changes:
1. The `BroadcastChannelService` class constructor has been modified by removing the `NavigationManager` parameter. This change simplifies the constructor and removes the dependency on the `NavigationManager`.
2. The `path` variable setting has been changed. Instead of using the `BaseUri` property of the `NavigationManager` instance, it is now directly set to the string `./_content/Blazor.BroadcastChannel/Blazor.BroadcastChannel.js`. This change makes the `path` setting more straightforward and less dependent on other variables.
3. A commented out line of code that sets the `_moduleTask` variable has been removed. The `_moduleTask` is still set in the same way, but now it uses the `path` variable instead of a hardcoded string. This change makes the code cleaner and easier to understand.

References to the code changes can be found in the `BroadcastChannelService` class and the method where the `path` variable is set.
